libtock/usb_keyboard_hid.c

```diff
@@ -0,0 +1,229 @@
+    *key = ((c -= '0') + 0x1D);
+    return 0;
+  }
+  switch (c) {
+    case '!':   *modifier = shift;
+      *key = 0x1E;
+      return 0;
+    case '@':   *modifier = shift;
+      *key = 0x1F;
+      return 0;
+    case '#':   *modifier = shift;
+      *key = 0x20;
+      return 0;
+    case '$':   *modifier = shift;
+      *key = 0x21;
+      return 0;
+    case '%':   *modifier = shift;
+      *key = 0x22;
+      return 0;
+    case '^':   *modifier = shift;
+      *key = 0x23;
+      return 0;
+    case '&':   *modifier = shift;
+      *key = 0x24;
+      return 0;
+    case '*':   *modifier = shift;
+      *key = 0x25;
+      return 0;
+    case '(':   *modifier = shift;
+      *key = 0x26;
+      return 0;
+    case ')':   *modifier = shift;
+      *key = 0x27;
+      return 0;
+    case '0':   *key = 0x27;
+      return 0;
+    case '\n':  *key = 0x28;
+      return 0;                          // enter
+    case '\r':  *key = 0x28;
+      return 0;                          // enter
+    case '\b':  *key = 0x2A;
+      return 0;                          // backspace
+    case '\t':  *key = 0x2B;
+      return 0;                          // tab
+    case ' ':   *key = 0x2C;
+      return 0;                          // space
+    case '_':   *modifier = shift;
+      *key = 0x2D;
+      return 0;
+    case '-':   *key = 0x2D;
+      return 0;
+    case '+':   *modifier = shift;
+      *key = 0x2E;
+      return 0;
+    case '=':   *key = 0x2E;
+      return 0;
+    case '{':   *modifier = shift;
+      *key = 0x2F;
+      return 0;
+    case '[':   *key = 0x2F;
+      return 0;
+    case '}':   *modifier = shift;
+      *key = 0x30;
+      return 0;
+    case ']':   *key = 0x30;
+      return 0;
+    case '|':   *modifier = shift;
+      *key = 0x31;
+      return 0;
+    case '\\':   *key = 0x31;
+      return 0;
+    case ':':   *modifier = shift;
+      *key = 0x33;
+      return 0;
+    case ';':   *key = 0x33;
+      return 0;
+    case '"':   *modifier = shift;
+      *key = 0x34;
+      return 0;
+    case '\'':   *key = 0x34;
+      return 0;
+    case '~':   *modifier = shift;
+      *key = 0x35;
+      return 0;
+    case '`':   *key = 0x35;
+      return 0;
+    case '<':   *modifier = shift;
+      *key = 0x36;
+      return 0;
+    case ',':   *key = 0x36;
+      return 0;
+    case '>':   *modifier = shift;
+      *key = 0x37;
+      return 0;
+    case '.':   *key = 0x37;
+      return 0;
+    case '?':   *modifier = shift;
+      *key = 0x38;
+      return 0;
+    case '/':   *key = 0x38;
+      return 0;
+  }
+  return -1;
+}
+
+
+
+int usb_keyboard_hid_send_letter_sync(uint8_t* buffer, char letter) {
+  int err;
+
+  uint8_t modifier;
+  uint8_t key;
+
+  to_hid_keycode(letter, &modifier, &key);
+
+  buffer[0] = modifier;
+  buffer[1] = 0;
+  buffer[2] = key;
+  buffer[3] = 0;
+  buffer[4] = 0;
+  buffer[5] = 0;
+  buffer[6] = 0;
+  buffer[7] = 0;
+
+  // Send key press.
+  err = usb_keyboard_hid_send_sync(buffer, 64);
```

Comment:
Why 64 bytes long? Does `buffer` need to be 64 bytes long?

Reply:
Yes.

Reply:
The usb_hid capsule only works on 64 byte buffers.

Reply:
The documentation of `usb_keyboard_hid_send_sync()` doesn't mention anything about 64 bytes though

Reply:
Does the capsule only work with 64 bytes? The documentation says at least 64 bytes